cdi/src/main/java/org/hibernate/validator/internal/cdi/ValidationExtension.java

```diff
@@ -212,11 +209,7 @@ else if ( bean.getTypes().contains( Validator.class ) || bean instanceof Validat
 	 * @param processAnnotatedTypeEvent event fired for each annotated type
 	 * @param <T> the annotated type
 	 */
-	public <T> void processAnnotatedType(@Observes @WithAnnotations({
```

Comment:
I think this might go a bit too far. Using `@WithAnnotations` is intentional here to limit the number of classes to check for automatic method validation. By removing it every annotated type will be processed and checked for constraints. I think we (Bean Validation) even were one of the driving forces on making the behavior of `@WithAnnotations` recursive. 

I think we need to dig deeper here and reach out to the Weld guys as well. It might be a bug in Weld and should be fixed there.